libs/dal/src/repositories/message/message.repository.ts

```diff
@@ -794,10 +832,10 @@ export class MessageRepository extends BaseRepository<MessageDBModel, MessageEnt
       );
     } else {
       // For non-seen operations, just do the regular update
-      await this.update(query, { $set: updatePayload });
+      await this.update(idQuery, { $set: updatePayload });
     }
 
-    return this.find(query);
```

Comment:
this one was a bug on our end because it was not returning the actual updated entries. 
(we can not fetch by the same query after we manipulated it)